scripts/stac/tests/generate_title_test.py

```diff
@@ -103,27 +103,30 @@ def test_generate_title_geographic_description(metadata: Tuple[CollectionMetadat
 
 def test_generate_title_event_imagery(metadata: Tuple[CollectionMetadata, CollectionMetadata]) -> None:
     _, metadata_hb = metadata
+    metadata_hb["geographic_description"] = "Hawke's Bay Cyclone Gabrielle"
     metadata_hb["event_name"] = "Cyclone Gabrielle"
     collection = ImageryCollection(metadata_hb)
-    title = "Hawke's Bay 0.3m Cyclone Gabrielle Rural Aerial Photos (2023)"
+    title = "Hawke's Bay Cyclone Gabrielle 0.3m Rural Aerial Photos (2023)"
     assert collection.stac["title"] == title
 
 
 def test_generate_title_event_elevation(metadata: Tuple[CollectionMetadata, CollectionMetadata]) -> None:
     _, metadata_hb = metadata
     metadata_hb["category"] = "dsm"
+    metadata_hb["geographic_description"] = "Hawke's Bay Cyclone Gabrielle"
     metadata_hb["event_name"] = "Cyclone Gabrielle"
     collection = ImageryCollection(metadata_hb)
-    title = "Hawke's Bay - Cyclone Gabrielle LiDAR 0.3m DSM (2023)"
+    title = "Hawke's Bay - Hawke's Bay Cyclone Gabrielle LiDAR 0.3m DSM (2023)"
```

Comment:
This doesn't look nice... I could change it to:
```python
if event:
    title = "region - event LiDAR..."
   # Hawke's Bay - Cyclone Gabrielle LiDAR 1m DSM (2023)
   # Nelson - Top of the South Floods LiDAR 1m DSM (2023)
elif geographic_description:
    title = "region - geographic_description LiDAR..."
    # Wellington - Hutt City LiDAR 1m DEM (2022)
title = "region LiDAR..."
# Taranaki LiDAR 1m DEM (2020)
```

Reply:
Sounds good, although I'd probably use an `else` for the fallback.

Reply:
I think we have to keep it to whatever geographic_description is added because I can think of scenarios where this would be worse.

The event name is clearly Cyclone Gabrielle but you might receive two different datasets within the same region and want to name one Hawke's Bay - Northern Rivers Cyclone Gabrielle and one Hawke's Bay - Southern Rivers Cyclone Gabrielle. You can't just use region and event name to distinguish those. But you can keep our STAC event field as Cyclone Gabrielle and then use "Northern Rivers Cyclone Gabrielle" and "Southern Rivers Cyclone Gabrielle" as the geographic_description.

The geographic_description will be a bit of a junk field on the STAC side of things because it will contain all sorts, but that's just what I feel is required in order to be able to name things in the most helpful way.